presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/NativeExecutionConnectorConfig.java

```diff
@@ -29,7 +28,8 @@
     private static final String CONNECTOR_NAME = "connector.name";
 
     private boolean cacheEnabled;
-    private DataSize maxCacheSize = new DataSize(0, DataSize.Unit.GIGABYTE);
+    // maxCacheSize in MB
+    private Long maxCacheSize = 0L;
```

Comment:
why  convert this to Long (Object) from DataSize ? Keeping it as DataSize, lets you write in Config 1MB and it gets parsed automatically. Converting it to long looses all those niceties and it is not presto convention.

Reply:
This is to be compatible with the C++ side which requires a numeric value rather than a size with unit.

Reply:
I am confused, this is Java code change here, can Java code read it using the DataSize and while sending to C++, can it send in the long format ?

Reply:
Make sense, I made the change that store the value in DataSize at Java, and only convert it to Long in the `getProperties()` method, the values returned by the `getProperties()`  will be sent to C++.